crates/kilt-dip-primitives/src/verifier/relaychain/error.rs

```diff
@@ -0,0 +1,80 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use crate::Error;
+
+#[cfg_attr(test, derive(enum_iterator::Sequence))]
+pub enum DipRelaychainStateProofVerifierError<DidOriginError> {
+	UnsupportedVersion,
+	ProofComponentTooLarge(u8),
+	ProofVerification(Error),
+	DidOriginError(DidOriginError),
+	Internal,
+}
+
+impl<DidOriginError> From<DipRelaychainStateProofVerifierError<DidOriginError>> for u16
+where
+	DidOriginError: Into<u8>,
+{
+	fn from(value: DipRelaychainStateProofVerifierError<DidOriginError>) -> Self {
+		match value {
+			// DO NOT USE 0
+			// Errors of different sub-parts are separated by a `u8::MAX`.
+			// A value of 0 would make it confusing whether it's the previous sub-part error (u8::MAX)
+			// or the new sub-part error (u8::MAX + 0).
+			DipRelaychainStateProofVerifierError::UnsupportedVersion => 1,
+			DipRelaychainStateProofVerifierError::ProofComponentTooLarge(component_id) => {
+				u8::MAX as u16 + component_id as u16
+			}
+			DipRelaychainStateProofVerifierError::ProofVerification(error) => {
+				u8::MAX as u16 * 2 + u8::from(error) as u16
+			}
+			DipRelaychainStateProofVerifierError::DidOriginError(error) => u8::MAX as u16 * 3 + error.into() as u16,
```

Comment:
This is already mentioned in [crates/kilt-dip-primitives/src/verifier/parachain/error.rs](https://github.com/KILTprotocol/kilt-node/pull/611/files/137bbdee9a9836b07d6610e4ce6644f7ba0c493d#diff-18cf44cea90aa9c135193ca60882f10a96ed14c27e8e2d32549b8496fb30501d).

Based on the result of the discussion there, this should be maybe updated too.